Support of MongoDB: Unified Topology Design

viktorzavadil opened the pull request:
I have added support of `useUnifiedTopology` option for MongoDB connection.

Feature requirements:
- you must provide own instance of MongoDB, see https://github.com/agenda/agenda#mongodbinstance
  ```js
  const client = new MongoClient(mongoCfg, {
    useNewUrlParser: true,
    useUnifiedTopology: true
  });
  
  await client.connect();
  
  const agenda = new Agenda({ mongo: client.db() });
  ```

Test of this feature is included.

simison commented on package.json:
```diff
@@ -63,7 +63,7 @@
     "jsdoc": "3.6.3",
     "jsdoc-template": "https://github.com/braintree/jsdoc-template",
-    "mocha": "6.2.2",
+    "mocha": "7.0.1",
```
Why the Mocha update in this PR?

viktorzavadil replied:
Ou, it seems `renovate-bot` update some dependencies. Maybe I used a bad branch name and that is the reason. What are the conventions?

viktorzavadil commented:
@simison Should I fix the linter issues, downgrade mocha to 6.2.2, rename the branch and create new PR?

viktorzavadil commented on test/connection-use-unified-topology.js:
```diff
@@ -0,0 +1,61 @@
+'use strict';
+/* eslint object-curly-spacing: "off" */
+/* eslint space-before-function-paren: "off" */
+/* eslint prefer-arrow-callback: "warn"  */
```
Yeah, it works.

I have different IDE settings and .editorconfig for this project is not enough for overriding all rules.

simison replied:
I can be useful to have Eslint integration installed in your editor directly, it should pick the right config in this folder I think. :-)

simison commented:
You can rebase your branch against `origin/master` and force push to it, that should get things sortet without needing to create new PR.

viktorzavadil commented:
@simison I have made a classic push into the old branch without rebasing and force push. What is the difference for you? If this is a problem I can try it but it seems like overkill :)

I fixed:
- mocha version,
- switched off linter rules.

simison commented:
@viktorzavadil looks better! Can you please also take out all `yarn.lock` changes in this PR? Thanks!

viktorzavadil commented:
@simison I forgot to change yarn changes after `renovate-bot` attack :smiley: I cherry-picked previous yarn version and I hope it will be good. I don't use yarn so I am not able to test it.

simison commented:
It's fine, Yarn is run in CI and tested there. 👍

MichielDeMey commented:
PR looks much cleaner now. 👍
